src/Orleans.Runtime/Utilities/UnobservedExceptionsHandler.cs

```diff
@@ -0,0 +1,90 @@
+using System;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Orleans.Runtime.Scheduler;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Orleans.Runtime
+{
+    internal static class ServiceProviderExtensions
```

Comment:
Maybe you could find a name related to SiloUnobservedExceptionsHandler, because this name is very generic and we might want to use it later in a broader context.

But it is mostly nitpicking here, no need to fix that if you don't want to